compiler/parser/src/conversions.rs

```diff
@@ -679,12 +688,91 @@ pub fn to_expression(node: &SyntaxNode<'_>, builder: &NodeBuilder, handler: &Han
+            {
+                // We use `unresolved` here because a `Get` might refer to `Vector::get` or
+                // `Mapping::get`. No way to know just yet.
+                leo_ast::AssociatedFunctionExpression {
+                    variant: leo_ast::Identifier::new(Symbol::intern("unresolved"), builder.next_id()),
+                    name,
+                    arguments: std::iter::once(receiver).chain(args).collect(),
+                    span,
+                    id: builder.next_id(),
+                }
+                .into()
+            } else if let (2, Some(leo_ast::CoreFunction::Set)) =
+                (args.len(), leo_ast::CoreFunction::from_symbols(Symbol::intern("unresolved"), name.name))
+            {
+                // We use `unresolved` here because a `Set` might refer to `Vector::set` or
+                // `Mapping::set`. No way to know just yet.
+                leo_ast::AssociatedFunctionExpression {
+                    variant: leo_ast::Identifier::new(Symbol::intern("unresolved"), builder.next_id()),
+                    name,
+                    arguments: std::iter::once(receiver).chain(args).collect(),
+                    span,
+                    id,
+                }
+                .into()
+            } else if let (1, Some(leo_ast::CoreFunction::VectorPush)) =
+                (args.len(), leo_ast::CoreFunction::from_symbols(sym::Vector, name.name))
+            {
+                leo_ast::AssociatedFunctionExpression {
+                    variant: leo_ast::Identifier::new(sym::Vector, builder.next_id()),
+                    name,
+                    arguments: std::iter::once(receiver).chain(args).collect(),
+                    span,
+                    id: builder.next_id(),
+                }
+                .into()
+            } else if let (0, Some(leo_ast::CoreFunction::VectorLen)) =
+                (args.len(), leo_ast::CoreFunction::from_symbols(sym::Vector, name.name))
+            {
+                leo_ast::AssociatedFunctionExpression {
+                    variant: leo_ast::Identifier::new(sym::Vector, builder.next_id()),
+                    name,
+                    arguments: vec![receiver],
+                    span,
+                    id: builder.next_id(),
+                }
+                .into()
+            } else if let (0, Some(leo_ast::CoreFunction::VectorPop)) =
+                (args.len(), leo_ast::CoreFunction::from_symbols(sym::Vector, name.name))
+            {
+                leo_ast::AssociatedFunctionExpression {
+                    variant: leo_ast::Identifier::new(sym::Vector, builder.next_id()),
+                    name,
+                    arguments: vec![receiver],
+                    span,
+                    id: builder.next_id(),
+                }
+                .into()
+            } else if let (0, Some(leo_ast::CoreFunction::VectorClear)) =
```

Comment:
Reminder to add a mention of clear to the PR description.

Reply:
Done